larpandoracontent/LArObjects/LArCaloHit.h

```diff
@@ -271,20 +308,36 @@ inline pandora::StatusCode LArCaloHitFactory::Read(Parameters &parameters, pando
     // ATTN: To receive this call-back must have already set file reader mc particle factory to this factory
     unsigned int larTPCVolumeId(std::numeric_limits<unsigned int>::max());
     unsigned int daughterVolumeId(0);
+    unsigned int nLabels(0);
+    std::vector<float> hitScores;
+    std::vector<std::string> hitScoreLabels;
 
     if (pandora::BINARY == fileReader.GetFileType())
     {
         pandora::BinaryFileReader &binaryFileReader(dynamic_cast<pandora::BinaryFileReader &>(fileReader));
         PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileReader.ReadVariable(larTPCVolumeId));
         if (m_version > 1)
             PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileReader.ReadVariable(daughterVolumeId));
+        if (m_version > 2) {
+            PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileReader.ReadVariable(hitScores));
```

Comment:
This won't work. You'll need to do something similar to what you've done with hitScoreLabels. Under the hood, ReadVariable checks the size of the type being passed in, but for a std::vector, this will not factor in the space allocated to the elements, only the address, current size and capacity information. The safest option here is to have an integer representing the number of elements in the vector that's stored in the file, and then read in N floats and populate the vector that way.